tests/arch/arm/arm_mpu_wt/src/main.c

```diff
@@ -0,0 +1,526 @@
+ */
+ZTEST(arm_mpu_wt, test_wt_performance)
+{
+	volatile uint32_t *test_addr;
+	uint32_t start_cycles, end_cycles, total_cycles;
+	const int iterations = 1000;
+
+	test_addr = alloc_aligned_memory(sizeof(uint32_t));
+	if (test_addr == NULL) {
+		ztest_test_skip();
+		return;
+	}
+
+	LOG_INF("Testing Write-Through performance characteristics");
+
+	/* Sequential write performance test */
+	start_cycles = k_cycle_get_32();
+
+	for (int i = 0; i < iterations; i++) {
+		*test_addr = i;
+		barrier_dsync_fence_full();
+	}
+
+	end_cycles = k_cycle_get_32();
+	total_cycles = end_cycles - start_cycles;
+
+	LOG_INF("Write performance: %u cycles for %d writes", total_cycles, iterations);
+	if (total_cycles > 0) {
+		LOG_INF("Average: %u cycles per write", total_cycles / iterations);
+	}
+
+	/* Verify final value */
+	zassert_equal(*test_addr, iterations - 1, "Final write value incorrect");
+
+	/* Sequential read performance test */
+	start_cycles = k_cycle_get_32();
+
+	volatile uint32_t dummy;
+
+	for (int i = 0; i < iterations; i++) {
+		dummy = *test_addr;
+		(void)dummy; /* Prevent compiler optimization */
+	}
+
+	end_cycles = k_cycle_get_32();
+	total_cycles = end_cycles - start_cycles;
+
+	LOG_INF("Read performance: %u cycles for %d reads", total_cycles, iterations);
+	if (total_cycles > 0) {
+		LOG_INF("Average: %u cycles per read", total_cycles / iterations);
+	}
+
+	k_free((void *)test_addr);
+}
+
+/**
+ * @brief Test Write-Through with stack memory
+ */
+ZTEST(arm_mpu_wt, test_wt_stack_access)
+{
+	uint32_t test_data[TEST_PATTERN_COUNT];
+	volatile uint32_t *test_addr = test_data;
+
+	LOG_INF("Testing Write-Through with stack memory at %p", (void *)test_addr);
+
+	/* Test each pattern */
+	for (int i = 0; i < TEST_PATTERN_COUNT; i++) {
+		test_addr[i] = test_patterns[i];
+		barrier_dsync_fence_full();
+
+		uint32_t read_value = test_addr[i];
+
+		zassert_equal(read_value, test_patterns[i],
+			      "Stack access failed at index %d: wrote 0x%08x, read 0x%08x",
+			      i, test_patterns[i], read_value);
+	}
+
+	LOG_INF("Stack memory test completed successfully");
+}
+
+#ifdef CONFIG_USERSPACE
+/**
+ * @brief Test Write-Through functionality in user mode
+ */
+ZTEST_USER(arm_mpu_wt, test_wt_userspace)
+{
+	volatile uint32_t *test_addr;
+	const uint32_t test_value = 0x12345678;
+
+	test_addr = alloc_aligned_memory(sizeof(uint32_t));
+	if (test_addr == NULL) {
+		ztest_test_skip();
+		return;
+	}
+
+	LOG_INF("Testing Write-Through in user mode at %p", test_addr);
+
+	/* Basic read/write test in user mode */
+	*test_addr = test_value;
+	barrier_dsync_fence_full();
+
+	uint32_t read_value = *test_addr;
+
+	zassert_equal(read_value, test_value,
+		      "User mode access failed: expected 0x%08x, got 0x%08x",
+		      test_value, read_value);
+
+	/* Test multiple patterns in user mode */
+	for (int i = 0; i < TEST_PATTERN_COUNT; i++) {
+		*test_addr = test_patterns[i];
+		barrier_dsync_fence_full();
```

Comment:
I do not thing it is good if we already have a cache system, and we need resync cache every time.